### PR TITLE
Updating docs in case of nested expressions

### PR DESCRIPTION
**Pull request by vtxorxwitty**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 1]
```diff
@@ -409,6 +409,15 @@ If an expression does not satisfy this condition, use the `toString()` [method](
 
 For example, in `/tmp/spe/<+pipeline.sequenceId>` the variable `sequenceId` evaluates to an integer. When concatenating this with other string expressions, it must be converted to a string, such as: `/tmp/spe/<+pipeline.sequenceId.toString()>`.
 
+If nested expressions are unable to resolve at some point during resolution, we return the concatenated string value of the partially resolved expression till that point as opposed to null in case of unnested expressions. For example:
```
Comment (sahilhindwani-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:412]: Can we be more clear here? I think this is not completely clear. 
What are nested expressions and unnested expressions? Can we give some examples?

Comment (vtxorxwitty) [reply docs/platform/variables-and-expressions/harness-variables.md:412]: Done

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 2]
```diff
@@ -409,6 +409,15 @@ If an expression does not satisfy this condition, use the `toString()` [method](
...
+
+```
+Assuming <+env.name> resolves to prod, and we don't have a pipeline variable prod defined - 
+Unnested Expression: <+pipeline.variables.prod>         --> null
```
Comment (sahilhindwani-harness) [inline docs/platform/variables-and-expressions/harness-variables.md:416]: This is not true, right? It depends on executionMode that the step is passing.

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]

File: docs/platform/variables-and-expressions/harness-variables.md (hunk 2 above)
Comment (vtxorxwitty) [reply docs/platform/variables-and-expressions/harness-variables.md:416]: Yes @sahilhindwani-harness , how would you want us to document this? Should we mention this is only is case of Shellscript. Or Should we document the executionMode corresponding to every step.

**Comment by bot2-harness** [comment]
This pull request has been automatically closed due to 30 days of inactivity. If you believe this was done in error, please feel free to reopen the PR or reach out to the BT team for further assistance. Thank you for your contributions!

**Pull request closed without merging**